File: Best-Time-to-Buy-and-Sell-Stock-Stock-Buy-and-Sell-Max-one-Transaction-Allowed-.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let max_p=0
    let mini=prices[0]
    for (let i=1;i<prices.length;i++){
        let cost=prices[i]-mini
        max_p=Math.max(max_p,cost)
        mini=Math.min(mini,prices[i])
    }
    return max_p
};

File: Best-Time-to-Buy-and-Sell-Stock-Stock-Buy-and-Sell-Max-one-Transaction-Allowed-.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_p=0
        mini=prices[0]
        for i in range(1,len(prices)):
            cost= prices[i]-mini
            max_p=max(cost,max_p)
            mini=min(mini,prices[i])
        return max_p
